tests/test_models.py

```diff
@@ -0,0 +1,487 @@
+# coding=utf-8
+import os
+import time
+import unittest
+
+from flask import g
+
+from app import create_app, db
+from app.api.custom_exceptions import (ValidationException,
+                                       UrlValidationException)
+from app.models import User, Url, ShortenUrl, AnonymousUser
+
+
+class UserModelTestCase(unittest.TestCase):
+    def setUp(self):
+        self.user = User(username="koyexes",lastname="koya",
+                         firstname="gabriel",email="koyexes@gmail.com")
+        self.user2 = User(username="admin",lastname="admin",
+                         firstname="admin",email="admin@gmail.com")
+        self.app = create_app('testing')
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        db.create_all()
+
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+    def test_password_setter(self):
+        """
+        checks if password setter sets the password property
+        appropriately based on the argument passed in.
+        """
+        self.user.password = "password"
+        self.assertTrue(self.user.password_hash is not None)
+
+    def test_hash_password(self):
+        """tests if password supplied was really hashed."""
+        self.user.password = "password"
+        self.assertFalse(self.user.password_hash == "password")
+
+    def test_no_password_getter(self):
+        """
+        tests the password getter if it raises the appropriate error
+        when invoke
+        """
+        self.user.password = "password"
+        with self.assertRaises(AttributeError):
+            self.user.password
+
+    def test_truthy_of_password_verification(self):
+        """
+        tests if  the true is returned for a correct password
+        passed in as argument
+        """
+        self.user.password = "password"
+        self.assertTrue(self.user.verify_password("password"))
+
+    def test_falsy_of_password_verification(self):
+        """
+        tests if  the false is returned for a correct password
+        passed in as argument
+        """
+        self.user.password = "password"
+        self.assertFalse(self.user.verify_password("wrong_password"))
+
+    def test_token_generation(self):
```

Comment:
I was actually testing whether a call to that method will return a value outside None, even thou the value returned can't be predicted.